components/portal/messages/filters.js

```diff
@@ -67,6 +67,40 @@ define(['angular'], function(angular) {
           }
         });
       };
+    })
+     .filter('filterMessagesWithInvalidDates', function() {
+      var retVal = [];
+      return function(messages) {
+          var thePresent = Date.now();
+          angular.forEach(messages, function(message) {
```

Comment:
:art: prefer `map` over `forEach + retVal + push` 

ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map
ref: https://code.tutsplus.com/tutorials/how-to-use-map-filter-reduce-in-javascript--cms-26209
ref: https://github.com/wix/eslint-plugin-lodash/blob/master/docs/rules/prefer-map.md

Reply:
It occurs to me we shouldn't be using `map` because we're not trying to transform/return every element. Rather, we're trying to filter out elements that don't match certain criteria, so wouldn't `Array.filter` be more appropriate? See #641

Reply:
Ah, I misread the data flow, `filter` would be more appropriate in this case.